django/econsensus/publicweb/tests/view_decision_test.py

```diff
@@ -46,3 +50,53 @@ def test_load_decision_form(self):
 
         form_data = self.get_form_values_from_response(response, 1)
         self.assertTrue(form_fields.issubset(set(form_data.keys())))
+
+    def test_decision_detail_contains_watch_toggle_link(self):
+        decision = self.create_and_return_decision()
+        doc = self._get_document('publicweb_decision_detail', decision.id)
+        elements = doc.cssselect('div#decision_detail a.watch.toggle')
+        self.assertEquals(len(elements), 1)
+
+    def test_view_page_watch_toggle_is_unwatched_when_no_user_is_watching(self):
+        decision = self.create_and_return_decision()
+        doc = self._get_document('publicweb_decision_detail', decision.id)
+        elements = doc.cssselect('div#decision_detail a.watch.toggle.unwatched')
+        self.assertEquals(len(elements), 1)
+
+    def test_view_page_watch_toggle_is_watched_when_the_current_user_is_watching(self):
+        decision = self.create_and_return_decision()
+        notification.observe(decision, self.user, DECISION_CHANGE)
+        doc = self._get_document('publicweb_decision_detail', decision.id)
+        elements = doc.cssselect('div#decision_detail a.watch.toggle.watched')
+        self.assertEquals(len(elements), 1)
+
+    def test_view_page_watch_toggle_is_unwatched_when_users_other_than_current_are_watching(self):
+        decision = self.create_and_return_decision()
+        notification.observe(decision, self.charlie, DECISION_CHANGE)
+        doc = self._get_document('publicweb_decision_detail', decision.id)
+        elements = doc.cssselect('div#decision_detail a.watch.toggle.unwatched')
+        self.assertEquals(len(elements), 1)
+
+    def test_following_unwatched_watch_toggle_adds_watcher(self):
```

Comment:
just wanted to say that i really like this test. not entirely sure why. but i do.